lib/versioning/nuget/index.spec.ts

```diff
@@ -1,60 +1,181 @@
 import nuget from '.';
 
 describe('nuget.', () => {
-  describe('isVersion and isValid', () => {
-    [
-      '9.0.3',
-      '1.2019.3.22',
-      '3.0.0-beta',
-      '2.0.2-pre20191018090318',
-      '1.0.0+c30d7625',
-      '2.3.4-beta+1990ef74',
-      '17.04',
-      '3.0.0.beta',
-      '5.1.2-+',
-    ].forEach((version) => {
-      it(version, () => {
-        expect(nuget.isVersion(version)).toMatchSnapshot();
-        expect(nuget.isValid(version)).toMatchSnapshot();
-      });
-    });
+  it('isValid', () => {
+    expect(nuget.isValid('9.0.3')).toBeTrue();
+    expect(nuget.isValid('9.0.3')).toBeTrue();
+    expect(nuget.isValid('1.2019.3.22')).toBeTrue();
+    expect(nuget.isValid('3.0.0-beta')).toBeTrue();
+    expect(nuget.isValid('2.0.2-pre20191018090318')).toBeTrue();
+    expect(nuget.isValid('1.0.0+c30d7625')).toBeTrue();
+    expect(nuget.isValid('2.3.4-beta+1990ef74')).toBeTrue();
+    expect(nuget.isValid('17.04')).toBeTrue();
+    expect(nuget.isValid('3.0.0.beta')).toBeFalse();
+    expect(nuget.isValid('5.1.2-+')).toBeFalse();
+    expect(nuget.isValid('[9.0.3]')).toBeTrue();
+    expect(nuget.isValid('(9.0.3)')).toBeFalse();
+    expect(nuget.isValid('9.0.3-*')).toBeTrue();
+    expect(nuget.isValid('9.0.*')).toBeTrue();
+    expect(nuget.isValid('9.*')).toBeTrue();
+    expect(nuget.isValid('*')).toBeTrue();
+    expect(nuget.isValid('9.*.3')).toBeFalse();
+    expect(nuget.isValid('1-a')).toBeTrue();
+    expect(nuget.isValid('1-b')).toBeTrue();
+    expect(nuget.isValid('[1-b,1-a]')).toBeFalse();
+    expect(nuget.isValid('[1-a,1-b]')).toBeTrue();
   });
-  describe('isStable', () => {
-    [
-      '9.0.3',
-      '1.2019.3.22',
-      '3.0.0-beta',
-      '2.0.2-pre20191018090318',
-      '1.0.0+c30d7625',
-      '2.3.4-beta+1990ef74',
-    ].forEach((version) => {
-      it(version, () => {
-        expect(nuget.isStable(version)).toMatchSnapshot();
-      });
-    });
+  it('isStable', () => {
+    expect(nuget.isStable('9.0.3')).toBeTrue();
```

Comment:
Didn't you say this represents a range, not a single version?

Reply:
Yes, for now I'm confused with this aspect since for we currently treat these values as single versions.

Reply:
I don't think there is a need for us to support functions like `isStable()` for strings for which `isSingleVersion()` returns false